tests/accessor_legacy/accessor_api_image_common.h

```diff
@@ -135,97 +112,44 @@ struct get<sycl::cl_half4> {
 };
 }
 
-/**
- * @brief Helper struct for retrieving the image access coordinates
- *        when testing an image array accessor
- * @tparam dims Number of accessor dimensions
- */
-template <int dims>
-struct image_array_coords {
-  /**
-   * @brief Retrieves the image access coordinates for an image array accessor
-   * @param idx Work-item ID
-   * @return Coordinates of dimension 1
-   */
-  template <typename ... rangeT>
-  static auto get(acc_coord_tag::use_int, image_array_id_t<dims> idx,
-                  rangeT ...)
-      -> typename image_access<dims>::int_type {
-    return image_access<dims>::get_int(
-        sycl_cts::util::get_cts_object::id<dims>::get(idx));
-  }
-  /**
-   * @brief Overload for cl_float type usage
-   */
-  template <typename ... rangeT>
-  static auto get(acc_coord_tag::use_float, image_array_id_t<dims> idx,
-                  rangeT...)
-      -> typename image_access<dims>::float_type {
-    return image_access<dims>::get_float(
-        sycl_cts::util::get_cts_object::id<dims>::get(idx));
-  }
-
-  /**
-   * @brief Overload for cl_float type usage with normalized coordinates
-   */
-  template <typename coordT,
-            typename pixelT = decltype(acc_coord_tag::get_pixel_tag(coordT{}))>
-  static auto get(coordT,
-                  image_array_id_t<dims> idx,
-                  image_array_range_t<dims> range)
-      -> typename image_access<dims>::float_type {
-    const auto resizedId =
-        sycl_cts::util::get_cts_object::id<dims>::get(idx);
-    const auto resizedRange =
-        sycl_cts::util::get_cts_object::range<dims>::get(range);
-
-    return image_access<dims>::get_normalized(pixelT{},
-                                              resizedId, resizedRange);
-  }
-};
-
 /**
  * @brief Constructs a range for testing
  * @tparam imageDims Number of image dimensions
  * @param count Number of elements to store in entire range
- * @param isImageArray True if creating a test range for an image array accessor
  * @return Range of same dimensions as the image
  */
 template <int imageDims>
-sycl::range<imageDims> make_test_range(size_t count,
-                                           bool isImageArray = false);
+sycl::range<imageDims> make_test_range(size_t count);
 
 /**
  * @brief Constructs a range for testing, specialization for 1D images
  * @param count Number of elements to store in entire range
  * @return 1D range
  */
 template <>
-sycl::range<1> make_test_range<1>(size_t count, bool /*isImageArray*/) {
+sycl::range<1> make_test_range<1>(size_t count) {
```

Comment:
It seems like the main goal of this PR is to remove image arrays, fixing the `value_type` is a very minor part of the changes here. I think it would be better to split image array removal into a separate PR, or at the very least make it obvious in the commit message and description.

Reply:
The PR was to fix a bug related to image_array, image, and host_image not being SYCL 2020 compliant, and their accessor functions being replaced by specialised alternatives. This change was a necessary part of fixing this bug.